src/payment/strategies/mollie/mollie-payment-strategy.ts

```diff
@@ -68,55 +79,23 @@ export default class MolliePaymentStrategy implements PaymentStrategy {
     async execute(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
         const { payment , ...order} = payload;
         const paymentData = payment?.paymentData;
-        const shouldSaveInstrument = (paymentData as HostedInstrument)?.shouldSaveInstrument;
-        const shouldSetAsDefaultInstrument = (paymentData as HostedInstrument)?.shouldSetAsDefaultInstrument;
 
-        if (!payment || !paymentData) {
-            throw new PaymentArgumentInvalidError([ 'payment' ]);
+        if (!payment || !payment.gatewayId || !paymentData) {
+            throw new PaymentArgumentInvalidError([ 'payment', 'gatewayId', 'paymentData' ]);
         }
 
         try {
-            if (payment.methodId === MolliePaymentMethodType.creditcard) {
-                await this._store.dispatch(this._orderActionCreator.submitOrder(order, options));
-
-                if (paymentData && isVaultedInstrument(paymentData)) {
-                    return this._store.dispatch(this._paymentActionCreator.submitPayment(payment));
-                }
-
-                const { token, error } = await this._getMollieClient().createToken();
+            await this._store.dispatch(this._orderActionCreator.submitOrder(order, options));
 
-                if (error) {
-                    return Promise.reject(error);
-                }
-
-                return await this._store.dispatch(this._paymentActionCreator.submitPayment({
-                    ...payment,
-                    paymentData: {
-                        formattedPayload: {
-                            credit_card_token: {
-                                token,
-                            },
-                            vault_payment_instrument: shouldSaveInstrument,
-                            set_as_default_stored_instrument: shouldSetAsDefaultInstrument,
-                            browser_info: getBrowserInfo(),
-                        },
-                    },
-                }));
-            } else {
-                await this._store.dispatch(this._orderActionCreator.submitOrder(order, options));
-
-                const issuer = 'issuer' in paymentData ? paymentData.issuer : '';
+            if (isVaultedInstrument(paymentData)) {
+                return this.executeWithVaulted(payment);
```

Comment:
I think you need to `await` these for the `try`/`catch` to work.

Reply:
Yes you were right! thanks!